### PR TITLE
demo(dialog): Add accessibility demo page for dialog

### PR DESCRIPTION
**Pull request by tinayuangao**

### CODE REVIEW

File: src/demo-app/a11y/dialog/dialog-a11y.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<section>
+  <h2>Welcome message</h2>
+  <p>Activat the button to see a welcome dialog with a simple message and a close button.</p>
```
Comment (jelbourn) [inline src/demo-app/a11y/dialog/dialog-a11y.html]: `Activat` -> `Activate`

File: src/demo-app/a11y/dialog/dialog-a11y.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+})
+export class DialogFruitExampleDialog {}
+
+@Component({
+  moduleId: module.id,
+  selector: 'dialog-welcome-a11y',
+  template: `Welcome to Angular Material dialog demo page!
+
+<md-dialog-actions>
+  <button
```
Comment (jelbourn) [inline src/demo-app/a11y/dialog/dialog-a11y.ts]: Should this dialog have some other content?

File: src/demo-app/a11y/dialog/dialog-neptune-a11y.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<h2 md-dialog-title>Neptune</h2>
+
+<md-dialog-content>
+  <img src="https://upload.wikimedia.org/wikipedia/commons/5/56/Neptune_Full.jpg"/>
```
Comment (jelbourn) [inline src/demo-app/a11y/dialog/dialog-neptune-a11y.html]: `img` needs an `alt`

File: src/demo-app/a11y/dialog/dialog-neptune-a11y.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+<md-dialog-actions [attr.align]="actionsAlignment">
+  <button md-raised-button
+          color="primary"
+          md-dialog-close>Close</button>
+
+  <a
+          md-button
+          color="primary"
+          href="https://en.wikipedia.org/wiki/Neptune"
+          target="_blank">Read more on Wikipedia</a>
```
Comment (jelbourn) [inline src/demo-app/a11y/dialog/dialog-neptune-a11y.html]: The indent / line-breaking here is odd

File: src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<h2 md-dialog-title>Neptune</h2>
```
Comment (jelbourn) [inline src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html]: Where does the focus go when this dialog is opened?

File: src/demo-app/a11y/dialog/dialog-a11y.ts (hunk 2 above)
Comment (tinayuangao) [inline src/demo-app/a11y/dialog/dialog-a11y.ts]: Do you have an example?

File: src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html (hunk 5 above)
Comment (tinayuangao) [inline src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html]: The "Close" button in the dialog.

File: src/demo-app/a11y/dialog/dialog-a11y.ts (hunk 2 above)
Comment (jelbourn) [inline src/demo-app/a11y/dialog/dialog-a11y.ts]: I was just thinking a `<p>` with some text

File: src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html (hunk 5 above)
Comment (jelbourn) [inline src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html]: I suspect that having the focus start _after_ all of the dialog content is weird

@tinayuangao @crisbeto what do you think of adding a dialog config option like `focusDialogElementOnOpen` (and also `ariaLabel`) to help with dialogs like this?

Comment (crisbeto) [inline src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html]: It does make sense, although specifying what element takes precedence in the focus trap should already be possible through the `cdk-focus-initial` attribute.

Comment (jelbourn) [inline src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html]: That doesn't give a way to focus the `role="dialog"` element, though, which I _think_ is what we'd want (but I'm not 100% sure).

Comment (tinayuangao) [inline src/demo-app/a11y/dialog/dialog-neptune-iframe-a11y.html]: Adding a config option makes sense to me. 

I added another example for forms. The focus goes to the first input box in the form when the dialog is opened.

**Comment by kara** [comment]
@tinayuangao Rebase?

**Comment by tinayuangao** [comment]
@kara Rebased. Thanks!

**Pull request merged**